dbms/src/Storages/DeltaMerge/tests/gtest_circular_scan_list.cpp

```diff
@@ -0,0 +1,123 @@
+#include <Storages/DeltaMerge/ReadThread/CircularScanList.h>
+#include <TestUtils/TiFlashTestBasic.h>
+
+#include <memory>
+
+namespace DB::DM::tests
+{
+class Node
+{
+public:
+    explicit Node(uint64_t id_)
+        : id(id_)
+        , table_id(1)
+        , v(true)
+    {}
+
+    bool valid() const
+    {
+        return v;
+    }
+    uint64_t poolId() const
+    {
+        return id;
+    }
+    int64_t tableId() const
+    {
+        return table_id;
+    }
+    void setInvalid()
+    {
+        v = false;
+    }
+
+private:
+    uint64_t id;
+    int64_t table_id;
+    bool v;
+};
+
+TEST(CircularScanListTest, Normal)
+{
+    CircularScanList<Node> lst;
+
+    {
+        ASSERT_EQ(lst.next(), nullptr);
+        auto [unvalid, valid] = lst.count(0);
+        ASSERT_EQ(unvalid, 0);
+        ASSERT_EQ(valid, 0);
+        ASSERT_EQ(lst.get(1), nullptr);
+    }
+
+    for (uint64_t i = 0; i < 10; i++)
+    {
+        lst.add(std::make_shared<Node>(i));
+    }
+
+    {
+        auto [unvalid, valid] = lst.count(0);
+        ASSERT_EQ(unvalid, 10);
+        ASSERT_EQ(valid, 0);
```

Comment:
The return value name should be `auto [valid, unvalid] = lst.count(0)`? ditto for later vars in this test
https://github.com/pingcap/tiflash/blob/2760de874b46abe4e9701995097f7bfa353230f7/dbms/src/Storages/DeltaMerge/ReadThread/CircularScanList.h#L58